### PR TITLE
Closes #439 league data store

### LINKED ISSUE #439: Connect league service database via Elide Multiplex store
The FAF league service stores it's data in a separate database on the same MySQL server.
Database host, username and password are the same, the database name is different

The tables are [here](https://github.com/FAForever/faf-league-service/blob/develop/migrations/V001__base.sql)

What we want to be able to query in Elide is:

- list of all leagues, seasons, division and subdivision (if possible leagues somehow with relationship to leaderboard)
- all score results of a player
- score changes per game
- all ranks of a player

### PR DESCRIPTION
**Pull request by bukajsytlos**
Closes #439 league data store

it's based on https://github.com/FAForever/faf-java-api/pull/433

- [x] permissions

### CODE REVIEW

**Review by ahsanbagwan — DISMISSED** [review]
LGTM

File: src/main/java/com/faforever/api/config/LeagueDatasourceConfig.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ) {
+    return builder
+      .dataSource(leagueDataSource)
+      .packages(
+        "com.faforever.api.league.domain"
+      )
+      .build();
+  }
+
+  @Bean
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/config/LeagueDatasourceConfig.java:50]: Shouldn't this need a name too?

File: src/main/java/com/faforever/api/config/elide/ElideConfig.java [hunk 2]
```diff
@@ -30,12 +29,20 @@
   public static final String DEFAULT_CACHE_NAME = "Elide.defaultCache";
 
   @Bean
-  public Elide elide(SpringHibernateDataStore springHibernateDataStore, ObjectMapper objectMapper, EntityDictionary entityDictionary, ExtendedAuditLogger extendedAuditLogger) {
+  MultiplexManager multiplexDataStore(
+    DataStore fafDataStore,
+    DataStore leagueDataStore
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/config/elide/ElideConfig.java:34]: No qualifiers? Why does this work?

File: src/main/java/com/faforever/api/league/domain/LeagueSeasonDivisionSubdivision.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Setter
+@Table(name = "league_season_division_subdivision")
+@Include(type = LeagueSeasonDivisionSubdivision.TYPE_NAME)
+public class LeagueSeasonDivisionSubdivision {
+  public static final String TYPE_NAME = "leagueSeasonDivisionSubdivisionV2";
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/league/domain/LeagueSeasonDivisionSubdivision.java]: Any reason for v2? We never launched a V1

File: src/main/java/com/faforever/api/config/elide/ElideConfig.java (hunk 2 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/config/elide/ElideConfig.java:34]: when the name of the bean factory method matches with parameter/field name it works(tm) most of the time :D

File: src/main/java/com/faforever/api/config/LeagueDatasourceConfig.java (hunk 1 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/config/LeagueDatasourceConfig.java:50]: name is the name of method

File: src/main/java/com/faforever/api/league/domain/LeagueSeasonDivisionSubdivision.java (hunk 3 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/league/domain/LeagueSeasonDivisionSubdivision.java]: renamed

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**